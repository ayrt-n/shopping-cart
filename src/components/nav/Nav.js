import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../images/logo.svg';
import '../../styles/Nav.css';

function Nav() {
  return (
    <nav className="Nav">
      <Link to="/">
        <img src={logo} className="Nav-logo" alt="Stussy logo" />
      </Link>
      <Link to="/shop" className="Nav-link">Shop</Link>
    </nav>
  );
}

export default Nav;
