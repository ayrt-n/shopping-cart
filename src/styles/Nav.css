.Nav {
  min-height: 3.25rem;
  padding: 0.75rem 1rem;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  position: fixed;
  background-color: white;
  z-index: 1;
}

.Nav-logo {
  height: 2.5rem;
  width: auto;
}

.Nav-link {
  text-decoration: none;
  color: black;
}

.Nav-link:hover {
  color: rgb(65, 65, 65);
}

.Nav-link:active {
  color: black;
}