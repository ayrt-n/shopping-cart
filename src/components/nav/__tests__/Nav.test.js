import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Nav from '../Nav';

describe('Nav component', () => {
  it('renders nav with links', () => {
    const { container } = render(<Nav />, {wrapper: MemoryRouter});
    expect(container).toMatchSnapshot();
  });
});
