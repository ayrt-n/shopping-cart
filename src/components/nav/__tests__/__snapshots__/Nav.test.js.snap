// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders nav with links 1`] = `
<div>
  <nav
    class="Nav"
  >
    <a
      href="/"
    >
      <img
        alt="Stussy logo"
        class="Nav-logo"
        src="test-file-stub"
      />
    </a>
    <a
      class="Nav-link"
      href="/shop"
    >
      Shop
    </a>
  </nav>
</div>
`;
